app/src/main/java/com/example/taskyy/data/local/notifications/NotificationScheduler.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.taskyy.data.local.notifications
+
+import android.app.AlarmManager
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.example.taskyy.ui.objects.AgendaEventItem
+
+class NotificationScheduler(private val context: Context) {
+    private val alarmManager = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+    @RequiresApi(Build.VERSION_CODES.S)
+    fun scheduleNotification(agendaItem: AgendaEventItem) {
+        alarmManager.apply {
+            canScheduleExactAlarms()
+            setExactAndAllowWhileIdle(
+                AlarmManager.RTC_WAKEUP,
+                agendaItem.alarmType,
+                createPendingIntent(agendaItem),
+            )
+        }
+    }
```

Comment:
And what do you do below Android S? This function only works above and will crash below